scilpy/tractograms/tractogram_operations.py

```diff
@@ -843,3 +854,246 @@ def split_sft_randomly_per_cluster(orig_sft, chunk_sizes, seed, thresholds):
     'concatenate': 'concatenate',
     'lazy_concatenate': 'lazy_concatenate'
 }
+
+
+def _compute_difference_for_voxel(chunk_indices,
+                                  skip_streamlines_distance=False):
+    """
+    Compute the difference between two sets of streamlines for a given voxel.
+    This function uses global variable to avoid duplicating the data for each
+    chunk of voxels.
+
+    Use the function tractogram_pairwise_comparison() as an entry point.
+
+    Parameters
+    ----------
+    chunk_indices: list
+        List of indices of the voxel to process.
+    skip_streamlines_distance: bool
+        If true, skip the computation of the distance between streamlines.
+
+    Returns
+    -------
+    results: list
+        List of the computed differences in the same order as the input voxel.
+    """
+    global sft_1, sft_2, matched_points_1, matched_points_2, tree_1, tree_2, \
+        sh_data_1, sh_data_2
+    results = []
+    for vox_ind in chunk_indices:
+        vox_ind = tuple(vox_ind)
+
+        global B
+        has_data = sh_data_1[vox_ind].any() and sh_data_2[vox_ind].any()
+        if has_data:
+            sf_1 = np.dot(sh_data_1[vox_ind], B)
+            sf_2 = np.dot(sh_data_2[vox_ind], B)
+            corr = np.corrcoef(sf_1, sf_2)[0, 1]
+        else:
+            corr = -1
+
+        if skip_streamlines_distance or not has_data:
+            results.append([-1, corr])
+            continue
+
+        # Get the streamlines in the neighborhood (i.e., 2mm away)
+        pts_ind_1 = tree_1.query_ball_point(vox_ind, 1.5)
```

Comment:
Your comment says 2mm, your code 1.5, the other comment below, 1. Should it be a variable? Even an option to user, or depending on the resolution of the data?